### DISCUSSION TITLE
Why do we need interactions ?

### DISCUSSION THREAD

**Discussion by Abdulllah-Rizwan** [Cyfrin/foundry-full-course-cu#643] [Q&A]
heya! Can anyone tell me in a most simplistic way that why do we need to make interaction file? what is the purpose of it?

**Comment by theirrationalone** [comment]
We want interaction between different number of functions to test whether one function respects another function or not in a specific routine/procedure. Sometimes we want to test it locally so that's why we have that `setUp` function that allows us to do interactions between functions in a protocol/contract locally. Next we have some separated interaction functions to test interactions between functions in any environment we want i.e., testnets(sepolia,goerli, koven, etc), mainnet(forked), even in local anvil too but spun by us not temporary.

So that's why interactions script plays an important role too.... 

 if we take our funMe contract's example.... 

There we first want to let `Funders` can deposit their funds with the respect of a specific condition(MINIMUM_USD)... Next we want that if contract/protocol has some balance funded by several funders then only after some funding we want let owner to withdraw.

So with having interactions script... we have a choice to simulate any procedure that we want to have step by step.... Imagine like... Do this after that, then after that do that and so on.

There is an another concept called INVARIANT(Fuzzing) where we run a automated campaign something a bit like interactions scripts but only for testing purposes.
